pythonforandroid/recipes/python2/__init__.py

```diff
@@ -5,41 +5,101 @@
                                        check_all, is_api_lt, is_ndk)
 from os.path import exists, join, realpath
 import sh
+import shutil
 
 
 class Python2Recipe(TargetPythonRecipe):
-    version = "2.7.2"
-    url = 'http://python.org/ftp/python/{version}/Python-{version}.tar.bz2'
+    version = "2.7.9"
+    url = 'http://python.org/ftp/python/{version}/Python-{version}.tgz'
     name = 'python2'
 
     depends = ['hostpython2']
     conflicts = ['python3crystax', 'python3']
-    opt_depends = ['openssl']
-    
-    patches = ['patches/Python-{version}-xcompile.patch',
-               'patches/Python-{version}-ctypes-disable-wchar.patch',
-               'patches/disable-modules.patch',
-               'patches/fix-locale.patch',
-               'patches/fix-gethostbyaddr.patch',
-               'patches/fix-setup-flags.patch',
+    opt_depends = ['openssl', 'libffi', 'sqlite3']
+
+    patches = ['patches/Python-{version}-xcompile_mod.patch',
+               'patches/Python_{version}-ctypes-libffi-fix-configure.patch',
+               'patches/ffi-config.sub-{version}.patch',
+               'patches/fix-locale-{version}.patch',
+               'patches/modules-locales-{version}.patch',
+               'patches/fix-platform-{version}.patch',
+               'patches/fix-gethostbyaddr.patch',  # OLD 2.7.2 Patch
+               'patches/basic-android-{version}_mod.patch',
+
+               # APPLY OLD WORKING 2.7.2 PATCHES
                'patches/fix-filesystemdefaultencoding.patch',
                'patches/fix-termios.patch',
                'patches/custom-loader.patch',
-               'patches/verbose-compilation.patch',
                'patches/fix-remove-corefoundation.patch',
                'patches/fix-dynamic-lookup.patch',
                'patches/fix-dlfcn.patch',
-               'patches/parsetuple.patch',
                'patches/ctypes-find-library-updated.patch',
-               ('patches/fix-configure-darwin.patch', is_darwin),
-               ('patches/fix-distutils-darwin.patch', is_darwin),
-               ('patches/fix-ftime-removal.patch', is_api_gt(19)),
-               ('patches/disable-openpty.patch', check_all(is_api_lt(21), is_ndk('crystax')))]
+
+               # Todo: Regular Patches from 2.7.2, Not needed on 2.7.9, must be removed?
```

Comment:
This requires more explanation. Why are these patches no longer necessary? Some of these are absolutely necessary, BTW.